container-runtime/src/Adapters/Systems/SystemForEmbassy/index.ts

```diff
@@ -335,6 +335,113 @@ export class SystemForEmbassy implements System {
     await this.exportActions(effects)
+    await this.exportNetwork(effects)
+  }
+  async exportNetwork(effects: HostSystemStartOs) {
+    for (const [id, interfaceValue] of Object.entries(
+      this.manifest.interfaces,
+    )) {
+      const hostId = `${id}-host`
+      for (const [external, internal] of Object.entries(
+        interfaceValue["tor-config"]?.["port-mapping"] ?? {},
+      )) {
+        const bindParams: T.BindParams = {
+          kind: "multi",
+          id: hostId,
+          internalPort: Number.parseInt(internal),
+          scheme: "http",
+          preferredExternalPort: Number.parseInt(external),
+          addSsl:
+            external === "443"
+              ? {
+                  scheme: "https",
+                  preferredExternalPort: Number.parseInt(external),
+                  alpn: "reflect",
+                }
+              : null,
+          secure:
+            external === "443"
+              ? {
+                  ssl: true,
+                }
+              : null,
+        }
+        await effects.bind(bindParams)
+      }
+      for (const [external, value] of Object.entries(
+        interfaceValue["lan-config"] ?? {},
+      )) {
+        const ssl = value.ssl
+        const internal = value.internal
+        const bindParams: T.BindParams = {
+          kind: "multi",
+          id: hostId,
+          internalPort: internal,
+          scheme: "http",
+          preferredExternalPort: Number.parseInt(external),
+          addSsl:
+            external === "443"
+              ? {
+                  scheme: "https",
+                  preferredExternalPort: Number.parseInt(external),
+                  alpn: "reflect",
+                }
+              : null,
+          secure: {
+            ssl: true,
+          },
+        }
+        await effects.bind(bindParams)
+      }
+      const options: T.ExportServiceInterfaceParams = {
+        id,
+        name: interfaceValue.name,
+        description: interfaceValue.description,
+        hasPrimary: id === "main" ? true : false,
+        disabled: false,
+        masked: false,
+        addressInfo: {
+          username: null,
+          hostId: hostId,
+          bindOptions: {
+            scheme: "http",
+            preferredExternalPort: 80,
+            addSsl: null,
+            secure: null,
+          },
+          suffix: "",
+        },
+        type: interfaceValue.ui ? "ui" : "api",
+        hostKind: "multi",
+        hostnames: [
+          ...Object.entries(interfaceValue["lan-config"] || {}).map(
+            ([key, value]): T.ExportedHostnameInfo => ({
+              kind: "ip",
+              networkInterfaceId: `${id}-lan-${key}`,
+              public: true,
+              hostname: {
+                kind: "local",
+                value: `${id}-lan-${key}-value`,
+                port: value.internal,
+                sslPort: null,
+              },
+            }),
+          ),
```

Comment:
this shouldn't be necessary - the os can get from the hostId on addressInfo

Reply:
Should be cleaned up now